genesyscloud/station/genesyscloud_station_proxy.go

```diff
@@ -0,0 +1,61 @@
+package station
+
+import (
+	"context"
+
+	"github.com/mypurecloud/platform-client-sdk-go/v115/platformclientv2"
+)
+
+// internalProxy holds a proxy instance that can be used throughout the package
+var internalProxy *stationProxy
+
+type getStationIdByNameFunc func(ctx context.Context, p *stationProxy, stationName string) (stationId string, retryable bool, err error)
+
+// stationProxy contains all of the methods that call genesys cloud APIs.
+type stationProxy struct {
+	clientConfig           *platformclientv2.Configuration
+	stationsApi            *platformclientv2.StationsApi
+	getStationIdByNameAttr getStationIdByNameFunc
+}
+
+// newStationProxy initializes the Station proxy with all of the data needed to communicate with Genesys Cloud
+func newStationProxy(clientConfig *platformclientv2.Configuration) *stationProxy {
+	stationsApi := platformclientv2.NewStationsApiWithConfig(clientConfig)
+
+	return &stationProxy{
+		clientConfig:           clientConfig,
+		stationsApi:            stationsApi,
+		getStationIdByNameAttr: getStationIdByNameFn,
+	}
+}
+
+// getStationProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getStationProxy(clientConfig *platformclientv2.Configuration) *stationProxy {
+	if internalProxy == nil {
+		internalProxy = newStationProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getStationIdByName retrieves a Genesys Cloud Station ID by Name
+func (p *stationProxy) getStationIdByName(ctx context.Context, stationName string) (stationId string, retryable bool, err error) {
+	return p.getStationIdByNameAttr(ctx, p, stationName)
+}
+
+// getStationIdByNameFn is an implementation function for retrieving a Station Id by Name
+func getStationIdByNameFn(ctx context.Context, p *stationProxy, stationName string) (stationId string, retryable bool, err error) {
+	const pageSize = 50
+	const pageNum = 1
+	stations, _, err := p.stationsApi.GetStations(pageSize, pageNum, "", stationName, "", "", "", "")
+	if err != nil {
+		return "", false, err
+	}
+
+	if stations.Entities == nil || len(*stations.Entities) == 0 {
+		return "", true, err
+	}
+
+	return *(*stations.Entities)[0].Id, false, nil
```

Comment:
It's possible for `stations` to contain multiple values. If the name in the method call was `abc` all stations with name containing `abc` will be returned such as `abc1` and `abc2`. You'll need to loop through `stations` and do an exact match on name

Reply:
fixed